lib/network/default.nix

```
@@ -0,0 +1,107 @@
+{ lib }:
+{
+  ipv6 = {
+    /**
+      Normalize an IPv6 address by removing the "::" compressions and padding
+      them with the necessary number of zeros to create the correct address. Converts
+      an address from the string to the list of strings which then can be parsed using
+      `_parseNormalized`. Throws an error when the address is malformed.
+
+      # Type: String -> [ String ]
+    */
+    _normalizeIpv6 =
+      addr:
+      if builtins.match "^[0-9A-Fa-f:]+$" addr == null then
+        throw "${addr} contains malformed characters for IPv6 address"
+      else
+        let
+          doubleColon = "::";
+          piecesMax = 8;
+          pieces = lib.strings.splitString ":" addr;
+          piecesNoEmpty = lib.lists.remove "" pieces;
+          piecesNoEmptyLen = builtins.length piecesNoEmpty;
+          emptyCnt = builtins.length pieces - builtins.length piecesNoEmpty;
+          hasPrefix = lib.strings.hasPrefix doubleColon addr;
+          hasSuffix = lib.strings.hasSuffix doubleColon addr;
+          hasInfix = lib.strings.hasInfix doubleColon addr;
```

I think we should keep it simple and just write "::" over introducing a extra binding, because it makes the code easier to read since you don't have to look up how doubleColon.

```suggestion
          hasPrefix = lib.strings.hasPrefix "::" addr;
          hasSuffix = lib.strings.hasSuffix "::"  addr;
          hasInfix = lib.strings.hasInfix "::" addr;
```